ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/visualization-selector/visualization-selector.js

```diff
@@ -0,0 +1,212 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ **/
+import React from 'react'
+import styled from '../../styles/styled-components'
+
+const CustomElement = styled.div`
+  height: 100%;
+  width: 100%;
+  display: block;
+`
+const Visualization = styled.div`
+  cursor: pointer;
+  opacity: ${props => props.theme.minimumOpacity}
+  :hover {
+    opacity: 1
+  }
+`
+const VisualizationChoice = styled.div`
+    white-space: nowrap;
+    padding: ${props => props.theme.largeSpacing}
+    cursor: move;
+    cursor: grab;
+    cursor: -moz-grab;
+    cursor: -webkit-grab;
+`
+
+const VisualizationIcon = styled.div`
+    text-align: center;
+    width: ${props => props.theme.minimumButtonSize}
+    display: inline-block;
+    vertical-align: middle;
+`
+const VisualizationText = styled.div`
+  width: calc(100% - ${props => props.theme.minimumButtonSize});
+  font-size: ${props => props.theme.mediumFontSize};
+  overflow: hidden;
+  text-overflow: ellipsis;
+  display: inline-block;
+  vertical-align: middle;
+`
+const configs = {
+  openlayers: {
+    title: '2D Map',
+    type: 'component',
+    componentName: 'openlayers',
+    componentState: {},
+  },
+  cesium: {
+    title: '3D Map',
+    type: 'component',
+    componentName: 'cesium',
+    componentState: {},
+  },
+  inspector: {
+    title: 'Inspector',
+    type: 'component',
+    componentName: 'inspector',
+    componentState: {},
+  },
+  table: {
+    title: 'Table',
+    type: 'component',
+    componentName: 'table',
+    componentState: {},
+  },
+  histogram: {
+    title: 'Histogram',
+    type: 'component',
+    componentName: 'histogram',
+    componentState: {},
+  },
+}
+
+const unMaximize = contentItem => {
+  if (contentItem.isMaximised) {
+    contentItem.toggleMaximise()
+    return true
+  } else if (contentItem.contentItems.length === 0) {
+    return false
+  } else {
+    return Array.some(contentItem.contentItems, subContentItem => {
+      return unMaximize(subContentItem)
+    })
+  }
+}
+
+class VisualizationSelector extends React.Component {
+  dragSources = []
```

Comment:
Any reason this is outside of state?

Reply:
https://github.com/codice/ddf/pull/4503/files#r265213803